src/extract_metadata.rs

```diff
@@ -1,26 +1,22 @@
-use exif::{DateTime, Reader, Tag, Value};
-use std::fs::File;
-use std::io::BufReader;
+use crate::date_time::DateTime;
 use std::path::Path;
+use crate::exiv2;
 
 #[derive(Debug)]
 pub struct PhotoMetadata {
     pub datetime: DateTime,
 }
 
 impl PhotoMetadata {
-    pub fn from_file(file_path: &Path) -> PhotoMetadata {
-        let file = File::open(file_path).unwrap();
-        let reader = Reader::new(&mut BufReader::new(&file)).unwrap();
-
-        let field = reader.get_field(Tag::DateTime, false).unwrap();
-        let data = match field.value {
-            Value::Ascii(ref vec) if !vec.is_empty() => vec[0],
-            _ => panic!(), // TODO: This should not panic, add error handling
-        };
-        let datetime = DateTime::from_ascii(data).unwrap();
-
-        PhotoMetadata { datetime }
+    pub fn from_file(file_path: &Path) -> Result<PhotoMetadata, exiv2::ExifReadError> {
+        let metadata = exiv2::read_metadata_from_file(file_path)?;
+        let date_time_str = &metadata["Exif"]["Image"]["DateTime"]
+            .as_str()
+            .ok_or_else(|| exiv2::ExifReadError {})?;
```

Comment:
napewno chcesz rzucać errorem kiedy nie ma daty? może jednak jako nullable i zostawianie jakiegoś warn-a + katalog dla zdjęć "do uporządkowania" jednocześnie żeby pastelogue zakończył robotę z tym co umiał

Reply:
Fajnie można by wtedy dodać zgłaszanie nieprawidłowości w działaniu ;)

Reply:
dobra chyba widzę, że faktycznie wtedy olewasz dalej przetwarzanie tego pliku, to jednak spoko wygląda :D

Reply:
Tak. Testowanie tego i może nawet lepsza obsługa zostawiam na kolejny PR. W tym PR mam tylko przepięcie zależności. Wymarzona obsługa błędów będzie w następnym PR.

Reply:
ok